### PR TITLE
bpo-45229: Make datetime tests discoverable

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue45229

### CODE REVIEW

File: Lib/test/test_datetime.py [hunk 1]
```diff
@@ -7,53 +7,52 @@
...
+            def tearDownClass(cls_):
+                sys.modules.clear()
+                sys.modules.update(cls_._save_sys_modules)
+            cls.setUpClass = setUpClass
+            cls.tearDownClass = tearDownClass
+            tests.addTests(loader.loadTestsFromTestCase(cls))
+    return tests
 
-def test_main():
-    run_unittest(*all_test_classes)
```
Comment (ambv) [inline Lib/test/test_datetime.py:56]: Since this isn't used anymore, the import of `run_unittest` isn't necessary either.

Comment (serhiy-storchaka) [reply Lib/test/test_datetime.py:56]: Good point, I missed this.

**Review by ambv — APPROVED** [review]
test_datetime on main: Ran 960 tests in 9.497s; OK (skipped=28)
test_datetime on this PR: Ran 960 tests in 9.468s; OK (skipped=28)

Same tests running:
```
--- /tmp/main_datetime.log	2021-09-29 21:26:34.000000000 +0200
+++ /tmp/28615_datetime.log	2021-09-29 21:28:25.000000000 +0200
@@ -1 +1 @@
-Wed Sep 29 21:26:23 CEST 2021
+Wed Sep 29 21:28:14 CEST 2021
@@ -3 +3 @@
-== CPython 3.11.0a0 (heads/main:45ca1c0413, Sep 29 2021, 21:24:05) [Clang 12.0.0 (clang-1200.0.32.29)]
+== CPython 3.11.0a0 (heads/test-datetime-discover:1131d9fb69, Sep 29 2021, 21:27:42) [Clang 12.0.0 (clang-1200.0.32.29)]
@@ -5 +5 @@
-== cwd: /Users/ambv/Dropbox/Python/cpython/build/test_python_40589æ
+== cwd: /Users/ambv/Dropbox/Python/cpython/build/test_python_49597æ
@@ -8,2 +8,2 @@
-0:00:00 load avg: 2.35 Run tests sequentially
-0:00:00 load avg: 2.35 [1/1] test_datetime
+0:00:00 load avg: 3.01 Run tests sequentially
+0:00:00 load avg: 3.01 [1/1] test_datetime
@@ -982 +982 @@
-Ran 960 tests in 9.497s
+Ran 960 tests in 9.468s
```

**Comment by pablogsal** [comment]
Unfortunately, this PR seems to have broken a bunch of buildbots. For instance:

https://buildbot.python.org/all/#/builders/75/builds/172/steps/5/logs/stdio

```
test_divide_and_round (test.datetimetester.TestModule_Pure) ... ok
test_divide_and_round (test.datetimetester.TestModule_Fast) ... skipped 'Only run for Pure Python implementation'
----------------------------------------------------------------------
Ran 2 tests in 0.000s
OK (skipped=1)
test_divide_and_round (test.datetimetester.TestModule_Pure) ... ERROR
.test test_datetime failed
test_divide_and_round (test.datetimetester.TestModule_Fast) ... skipped 'Only run for Pure Python implementation'
======================================================================
ERROR: test_divide_and_round (test.datetimetester.TestModule_Pure)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dje/cpython-buildarea/3.x.edelsohn-rhel8-z.refleak/build/Lib/test/datetimetester.py", line 88, in test_divide_and_round
    dar = datetime_module._divide_and_round
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'datetime' has no attribute '_divide_and_round'
----------------------------------------------------------------------
Ran 2 tests in 0.003s
FAILED (errors=1, skipped=1)
1 test failed again:
    test_datetime
== Tests result: FAILURE then FAILURE ==
```

**Pull request merged**